### ISSUE TITLE
Cabal 2's internal libraries don't work with HEAD

### ISSUE THREAD

**Issue by bucklereed** [commercialhaskell/stack#3361]
Cabal 2 support has just landed in HEAD with #3288 (hurrah), but one of the shiny new features doesn't work with stack yet: internal libraries (introduced in haskell/cabal#3022). A short demonstration:
```
$ cat internal-library-test.cabal 
name: internal-library-test
version: 1
cabal-version: >= 1.24

library internal-library-test-lib
  exposed-modules: Foo
  default-language: Haskell2010

executable exe
  main-is: Main.hs
  default-language: Haskell2010
  build-depends: internal-library-test-lib
$ cat stack.yaml 
resolver: lts-9.1
packages: ['.']
$ stack build

Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for internal-library-test-1:
    internal-library-test-lib must match -any, but the stack configuration has no specified version
needed since internal-library-test-1 is a build target.

Plan construction failed.
$ stack --version
Version 1.5.1, Git revision aa5003a153504f54051e42a844a0c1c3d7f82163 i386 hpack-0.17.1
```

I *think* that this ought to work as-is (that is, without anything internal-library-specific in stack.yaml), but it wouldn't be a disaster if it needed some magical incantation.

**Comment by snoyberg** [comment]
Not certain, but: it seems like a simple workaround for this _may_ be that, during the cabal file parsing step, to simply remove any build-depends referring to an internal library. I'm not sure if Cabal will then handle the build correctly from that point, but it's worth a shot.

**Comment by snoyberg** [comment]
This should be resolved by #3430. It would be great if you could test that branch before this gets merged to master.

**Comment by lwm** [comment]
Closing this one off - please re-open if #3430 doesn't solve it.
